src/freertos_drivers/spiffs/cc32x0sf/CC32x0SFSPIFFS.cxx

```diff
@@ -0,0 +1,147 @@
+/** @copyright
+ * Copyright (c) 2018, Stuart W Baker
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are  permitted provided that the following conditions are met:
+ * 
+ *  - Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ *  - Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * @file CC32x0SFSPIFFS.cxx
+ * This file implements a SPIFFS FLASH driver specific to CC32xx.
+ *
+ * @author Stuart W. Baker
+ * @date 1 January 2018
+ */
+
+#include "CC32x0SFSPIFFS.hxx"
+
+#include "inc/hw_types.h"
+#include "driverlib/flash.h"
+
+//
+// CC32x0SFSPIFFS::flash_read()
+//
+int32_t CC32x0SFSPIFFS::flash_read(uint32_t addr, uint32_t size, uint8_t *dst)
+{
+    HASSERT(addr >= config_.phys_addr &&
+            (addr + size) <= (config_.phys_addr  + config_.phys_size));
+
+    memcpy(dst, (void*)addr, size);
+
+    return 0;
+}
+
+//
+// CC32x0SFSPIFFS::flash_write()
+//
```

Comment:
This method is really ugly.  It "works", but I'm open to suggestions on how to simplify this.  I fear there may be hidden bugs here that will blow up at some point.

Reply:
This is not quite right either.  I filed an issue to look at this in more detail:  https://github.com/bakerstu/openmrn/issues/115